File: DataBindingObservableJava/app/src/main/java/com/websarva/wings/android/databindingobservablejava/MainActivity.java
package com.websarva.wings.android.databindingobservablejava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.websarva.wings.android.databindingobservablejava.databinding.ActivityMainBinding;
import com.websarva.wings.android.databindingobservablejava.ui.MainViewModel;

public class MainActivity extends AppCompatActivity {
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(MainActivity.this, R.layout.activity_main);
		ViewModelProvider provider = new ViewModelProvider(MainActivity.this);
		MainViewModel mainViewModel = provider.get(MainViewModel.class);
		activityMainBinding.setMainViewModel(mainViewModel);
	}
}

File: DataBinding2WayKotlin/app/src/main/java/com/websarva/wings/android/databinding2waykotlin/MainActivity.kt
package com.websarva.wings.android.databinding2waykotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import com.websarva.wings.android.databinding2waykotlin.databinding.ActivityMainBinding
import com.websarva.wings.android.databinding2waykotlin.ui.MainViewModel

class MainActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val activityMainBinding = DataBindingUtil.setContentView<ActivityMainBinding>(this@MainActivity, R.layout.activity_main)
		val mainViewModel by viewModels<MainViewModel>()
		activityMainBinding.mainViewModel = mainViewModel
	}
}
